src/hist/basehist.py

```diff
@@ -150,20 +161,140 @@ def project(self, *args: Union[int, str]):
         elif all(isinstance(x, str) for x in args):
             indices: tuple = tuple()
             for name in args:
-                for index, axis in enumerate(self.axes):
-                    if name == axis.name:
-                        indices += (index,)
-                        break
-                else:
-                    raise ValueError("The axis names could not be found")
+                if isinstance(
+                    name, str
+                ):  # ToDo: redundant line - mypy cannot realized it
```

Comment:
Then use:

```python
assert isinstance(name, str) # for MyPy
```

here. This will make mypy believe you.

Reply:
Or you can move the check down here, instead of at the top. That way, you don't loop over everything twice (as `all`/`any` do a loop inside).